aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2532,7 +2532,7 @@ Tensor transpose(const Tensor & self, int64_t dim0, int64_t dim1) {
 
   // Transpose of a tensor is a view operation.
   if (dim0 == dim1) {
-    return self;
+    return self.alias();
```

Comment:
This is breaking the SparseCSR test 

I am special case this for SparseCSR Tensor

    if (self.is_sparse_csr()) {
      return self.clone();
    } else {
      return self.alias();
    }

Reply:
Seems right, according to the other behavior in the file.

Reply:
Please don't incur a copy when it's actually not needed.

Reply:
I assumed that was what this was doing? https://github.com/pytorch/pytorch/pull/84674/files#diff-e70cddcca22b07fef2f931799e771dd0f462dac8f35877861137db498643e6cfR2523